src/auth_and_perms/api/serializers.py

```diff
@@ -162,6 +162,8 @@ def validate(self, data):
         laboratory = data['laboratory']
         organization = data['organization']
         user = self.context.get('user')
+        shelf_object = data["shelfobject"] if "shelfobject" in data else None
```

Comment:
here we can use a data.get('shelfobject') or data.get('shelf') - this way it will be None by default if it is not present, and it is better for performance I think.